crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -51,6 +51,13 @@ impl TEdgeComponent for CumulocityMapper {
         let mut uploader_actor = UploaderActor::new(identity.clone()).builder();
         let mut downloader_actor = DownloaderActor::new(identity).builder();
 
+        // MQTT client dedicated to monitor the c8y-bridge client status and also
+        // set service down status on shutdown, using a last-will message.
+        // A separate MQTT actor/client is required as the last will message of the main MQTT actor
+        // is used to send down status to health topic.
+        let mut service_monitor_actor =
+            MqttActorBuilder::new(service_monitor_client_config(&tedge_config)?);
```

Comment:
This MQTT actor which was previously used just to send the last will message is reused to monitor the bridge health status as well.